### PR TITLE
Handle exceptions thrown during checkAndInitialize

### PR DESCRIPTION
**Pull request by deepakala-k**
If dump manager crashes core file will be generated but core dump will not be created as the dump service that creates core dump is taking a reset. Wee check for core files at the start of the dump service and capture them in core dump.

During this scenario, if there is any exception with capturing the dump, it was not handled.

Fix:
Added exception handling for the above scenario.

### CODE REVIEW

**Comment by SwarnenduRC** [comment]
Typo; "we" became "wee". `if there is any exception with capturing the dump, it was not handled. `can be rephrased as "if there was an exception while capturing the dump it was unhandled`

File: dump_manager_bmc.cpp [hunk 1]
```diff
@@ -166,7 +166,20 @@ void Manager::checkAndCreateCoreDump()
             log<level::INFO>(
                 fmt::format("Core file found, files size {}", files.size())
                     .c_str());
-            captureDump(Type::ApplicationCored, files);
+
+            try
+            {
+                captureDump(Type::ApplicationCored, files);
+            }
+            catch (const std::exception& excp)
```
Comment (SwarnenduRC) [inline dump_manager_bmc.cpp:174]: Can we have derived class exception first followed by standard one? If possible then that would be more robust and verbose.

**Review by SwarnenduRC — CHANGES_REQUESTED** [review]
Few observations

**Comment by deepakala-k** [comment]
> if there was an exception while capturing the dump it was unhandled

Done.

File: dump_manager_bmc.cpp [hunk 2]
```diff
@@ -166,7 +169,29 @@ void Manager::checkAndCreateCoreDump()
...
+                captureDump(Type::ApplicationCored, files);
+            }
+            catch (const QuotaExceeded& e)
+            {
+                log<level::ERR>(
+                    fmt::format(
+                        "checkAndCreateCoreDump:: Exception Caught while capturing dump: "
+                        "{} REASON: {}",
+                        e.name(), e.description())
+                        .c_str());
```
Comment (dhruvibm) [inline dump_manager_bmc.cpp:184]: We should return here since no further dump creation is possible

**Review by dhruvibm — CHANGES_REQUESTED** [review]
one comment

File: dump_manager_bmc.cpp (hunk 2 above)
Comment (deepakala-k) [reply dump_manager_bmc.cpp:184]: Sure.

Comment (deepakala-k) [reply dump_manager_bmc.cpp:184]: But there is nothing after that. So anyways it will return. Do you still want a return statement here?

Comment (deepakala-k) [reply dump_manager_bmc.cpp:184]: Or would it better to handle the exception in the main? https://github.com/ibm-openbmc/phosphor-debug-collector/blob/1020/dump_manager_main.cpp#L98

Comment (dhruvibm) [reply dump_manager_bmc.cpp:184]: It is better to handle here, main doesn't need to know about the difference between other exceptions and quota exceeded,

**Review by dhruvibm — APPROVED** [review]

**Review by dhruvibm — COMMENTED** [review]
Need to fix CI

**Pull request merged**